StackExchange.Redis/StackExchange/Redis/GeoEntry.cs

```diff
@@ -0,0 +1,284 @@
+            this.longitude = longitude;
+            this.latitude = latitude;
+        }
+
+        /// <summary>
+        /// See Object.ToString()
+        /// </summary>
+        public override string ToString()
+        {
+            return string.Format("{0} {1}", longitude, latitude);
+        }
+        /// <summary>
+        /// See Object.GetHashCode()
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return longitude.GetHashCode() ^ latitude.GetHashCode();
+        }
+        /// <summary>
+        /// Compares two values for equality
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            return obj is GeoPosition && Equals((GeoPosition)obj);
+        }
+        /// <summary>
+        /// Compares two values for equality
+        /// </summary>
+        public bool Equals(GeoPosition value)
+        {
+            return Equals(longitude, value.Longitude) && Equals(latitude, value.latitude);
+        }
+        /// <summary>
+        /// Compares two values for equality
+        /// </summary>
+        public static bool operator ==(GeoPosition x, GeoPosition y)
+        {
+            return Equals(x.longitude, y.longitude) && Equals(x.latitude, y.latitude);
+        }
+        /// <summary>
+        /// Compares two values for non-equality
+        /// </summary>
+        public static bool operator !=(GeoPosition x, GeoPosition y)
+        {
+            return !Equals(x.longitude, y.longitude) || !Equals(x.latitude, y.latitude);
+        }
+    }
+
+    /// <summary>
+    /// Describes a GeoEntry element with the corresponding value
+    /// GeoEntries are stored in redis as SortedSetEntries
+    /// </summary>
+    public struct GeoEntry : IEquatable<GeoEntry>
+    {
+        internal readonly RedisValue member;
+        internal readonly GeoPosition geoPos;
+
+        /// <summary>
+        /// Initializes a GeoEntry value
+        /// </summary>
+        public GeoEntry(double longitude,double latitude,RedisValue member)
+        {
+            this.member = member;
+            geoPos = new GeoPosition(longitude, latitude);
+        }
+
+        /// <summary>
+        /// The name of the geo entry
+        /// </summary>
+        public string Member => member;
+
+        /// <summary>
+        /// The longitude of the geo entry
+        /// </summary>
+        public double Longitude => geoPos.Longitude;
+
+        /// <summary>
+        /// The latitude of the geo entry
+        /// </summary>
+        public double Latitude => geoPos.Latitude;
+
+        /// <summary>
+        /// Describes the longitude and latitude of a GeoEntry
+        /// </summary>
+        public GeoPosition GeoPosition => geoPos;
+       
+
+        /// <summary>
+        /// See Object.ToString()
+        /// </summary>
+        public override string ToString()
+       {
+           return string.Format("{0} {1} {2}", geoPos.Latitude, geoPos.Latitude, member);
```

Comment:
might be clearer to do something that looks a: co-ordinates, and b: key-value; for example `return $"({Longitude},{Latitude})={member}";` - also, note you use `Latitude` twice